src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -247,7 +246,7 @@ public void Add(T item)
             var array = _items;
             var size = _size;
             _version++;
-            if ((uint)size < (uint)array.Length)
+            if (array.IndexInRange(size))
```

Comment:
> Not sure what you have in mind. 

If signed test was performed, the range check just has to test its >= 0, rather than comparing to length.

> What's the cost of a bounds check anyway?

The examples I was looking at the generally always ended up as a memory access; so probably dominated by L1 cache hit? (as already checked) 
```asm
cmp      r8d, dword ptr [rax+8]
jae      SHORT G_M46198_IG06 
```
Maybe if it kept the length in register from earlier lookup and test that would be most of the cost gone.